packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -126,14 +139,37 @@ export class SelectedNetworkController extends BaseController<
       state,
     });
     this.#registerMessageHandlers();
+
+    // if the network for metamask is not set, set it to the currently selected network upon initialization
```

Comment:
Really the `metamask` domain selected network client should be identical to the NetworkController selected network client at all times. This only changes it if it's unset, but it's problematic if they're out of sync as well.

We never should have introduced this duplicate state in the first place IMO. If we do need it, we should treat one instance as the source of truth, and the other as derived state. Independent management of state that we assume to be identical is asking for trouble.

Reply:
> Independent management of state that we assume to be identical is asking for trouble.
Totally agree. I believe we had a conversation about this somewhere in slack (having trouble finding it) but basically I think the reason for this duplicate state is that it makes more sense for this state to live in this controller (the `SelectedNetworkController`, rather than the `NetworkController`) but we don't yet feel ready to remove the `selectedNetworkClientId` and `providerConfig` state from the `NetworkController`. I'm fine with removing this state from `SelectedNetworkController` until we're ready to move it over from the `NetworkController` . I honestly can't remember if there's any good reason not to just do that...

Reply:
I've got a commit here that cleans a few things up per your feedback including removing the duplication of the 'metamask' selected `networkClientId` into this controller: https://github.com/MetaMask/core/compare/ad/fix-selected-network-middleware...ad/fix-selected-network-middleware-tweaked?expand=1

Reply:
testing these updates here and seeing it working as expected: https://github.com/MetaMask/metamask-extension/pull/22860  🥳

Reply:
Yep. I'm going to merge in those changes and fix up tests accordingly.

Reply:
I've removed this derived state entirely here: https://github.com/MetaMask/core/pull/3908/commits/c78443841150f1f0acf3007fa9d67fae9bce5481

Reply:
Why go backwards here with relying on `selectedNetworkClientId`? I'd argue to remove `selectedNetworkClientId` from the NetworkController over removing `metamask` domain from `SelectedNetworkController` as the way forward.

Reply:
I don't disagree @shanejonas, except **I think** removing the `selectedNetworkClientId` from the `NetworkController` requires a bit more thought, in part because we still have a globally selected `providerConfig` that is widely used and will be complicated to extract. Because of this, if we move the `metamask` domain selectedNetworkClientId from the NetworkController right now, the globally selected state will be fragmented across two controllers. I'm thinking we keep that all in one controller until we've organized our thoughts for migrating it completely and use SelectedNetworkController **just** for all non-globally selected network state until then.

Reply:
Agreed with Alex's thoughts. I don't think it was necessarily wrong to have the globally selected networkClientId set as the metamask domain in this controller, but that leaves us with two sources of truth. The globally selected networkClientId will eventually be removed anyway, so we shouldn't get too hung up on if it belongs in this controller or not.